### PR TITLE
simplifying setUnitPlacementOnlyAllowedIn

### PR DESCRIPTION
**Pull request by veqryn**
[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/311)

### CODE REVIEW

**Comment by veqryn** [comment]
@DanVanAtta from #293

File: src/games/strategy/triplea/attatchments/UnitAttachment.java [hunk 1]
```diff
@@ -799,30 +798,16 @@ public void resetUnitPlacementRestrictions() {
...
-
-      String newValue = valueRestricted.toString().replaceFirst(":", "");
-      m_unitPlacementRestrictions = newValue.split(":");
-    }
+    final Collection<Territory> allowedTerritories = getListedTerritories(value.split(":"));
+    final Collection<Territory> restrictedTerritories = new HashSet<>(getData().getMap().getTerritories());
+    restrictedTerritories.removeAll(allowedTerritories);
+    m_unitPlacementRestrictions = restrictedTerritories.stream()
+        .map(Territory::getName)
+        .toArray(size -> new String[size]);
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/attatchments/UnitAttachment.java]: Mostly out of curiosity, do you need to set the array size here? I haven't used stream much yet so just wondering.

**Comment by ron-murhammer** [comment]
Just the one question above, otherwise this looks good.

File: src/games/strategy/triplea/attatchments/UnitAttachment.java (hunk 1 above)
Comment (veqryn) [inline src/games/strategy/triplea/attatchments/UnitAttachment.java]: I'm not too familiar with it yet either, but this is how the internet says to convert a stream to an array.

**Pull request merged**